types/estree/index.d.ts

```diff
@@ -251,16 +251,29 @@ export interface ObjectExpression extends BaseExpression {
   properties: Array<Property | SpreadElement>;
 }
 
+export interface PrivateIdentifier extends BaseNode {
+    type: "PrivateIdentifier";
+    name: string;
+}
+
 export interface Property extends BaseNode {
   type: "Property";
-  key: Expression;
+  key: Expression | PrivateIdentifier;
   value: Expression | Pattern; // Could be an AssignmentProperty
   kind: "init" | "get" | "set";
   method: boolean;
   shorthand: boolean;
   computed: boolean;
 }
 
+export interface PropertyDefinition extends BaseNode {
+    type: "PropertyDefinition";
+    key: Expression | PrivateIdentifier;
+    value: Expression | null;
+    computed: boolean;
```

Comment:
In the future we might want to do smarter types here (to only allow `computed: false` when `key` is `PrivateIdentifier` as per spec).

Reply:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/52832/commits/584217d1cb85f04e2b1c47c082b3212429bf457f